### PR TITLE
Finish user cancel order

### LINKED ISSUE #16: User Story 27, User cancels an order
```
As a registered user
When I visit an order's show page
I see a button or link to cancel the order only if the order is still pending
When I click the cancel button for an order, the following happens:
- Each row in the "order items" table is given a status of "unfulfilled"
- The order itself is given a status of "cancelled"
- Any item quantities in the order that were previously fulfilled have their quantities returned to their respective merchant's inventory for that item.
- I am returned to my profile page
- I see a flash message telling me the order is now cancelled
- And I see that this order now has an updated status of "cancelled"
```

DO THESE STORIES FIRST:
- User Story 18, User Profile Show Page
- User Story 26, User views an Order Show Page

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #222

child of #242

depends on #220

depends on #212

### PR DESCRIPTION
**Pull request by hillstew**
Closes #16

### CODE REVIEW

File: spec/features/orders/user_cancels_order_spec.rb [hunk 1]
```diff
@@ -2,8 +2,8 @@
 
 describe "As a registered user" do
   describe "when I visit an orders show page" do
-    xit "I click 'Cancel Order' button only if order is pending" do
```
Comment (tyladevon) [inline spec/features/orders/user_cancels_order_spec.rb]: Nice clean up in the Test file!

File: spec/features/orders/user_cancels_order_spec.rb [hunk 2]
```diff
@@ -14,22 +14,23 @@
...
+      visit "/profile/orders/#{order_1.id}"
       click_button "Cancel Order"
-
+     
       expect(current_path).to eq("/profile")
-      expect(page).to have_content("Your order is now cancelled")
-      expect(@order_1.status).to eq("Cancelled")
+      expect(page).to have_content("Your order has been cancelled")
+
+      updated_order = Order.find(order_1.id)
```
Comment (rhantak) [inline spec/features/orders/user_cancels_order_spec.rb]: This was tough! Good thinking having the test make a database call vs. checking an existing variable
